portfolio/src/main/java/com/google/sps/servlets/ContactFormServlet.java

```diff
@@ -0,0 +1,94 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import com.google.gson.Gson;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+
+/** Servlet that processes text. */
+@WebServlet("/contact_form")
+public final class ContactFormServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get input from form
+    String fname = getParameter(request, "fname", "");
+    String lname = getParameter(request, "lname", "");
+    String email = getParameter(request, "email", "");
+    String mobile = getParameter(request, "mobile", "");
+    String message = getParameter(request, "message", "");
+
+    // // Respond with the result.
+    // response.setContentType("text/html;");
+    // response.getWriter().println(form);
+
+    Entity form_entity = new Entity("contact_form");
+    form_entity.setProperty("fname", fname);
+    form_entity.setProperty("lname", lname);
+    form_entity.setProperty("email", email);
+    form_entity.setProperty("mobile", mobile);
+    form_entity.setProperty("message", message);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
should put the datastore initialization in init, so that you don't create a new datastore every time